packages/db/src/project/loadMigrate/contracts.ts

```diff
@@ -19,29 +19,41 @@ export const process = Batch.configure<{
     };
   };
   produces: {
-    callBytecode?: {
+    callBytecode: {
       linkReferences: { name: string }[];
     };
-    createBytecode?: {
+    createBytecode: {
       linkReferences: { name: string }[];
     };
   };
   entry: IdObject<"contracts">;
-  result: {
-    callBytecode?: {
-      linkReferences: { name: string }[];
-    };
-    createBytecode?: {
-      linkReferences: { name: string }[];
-    };
-  };
+  result:
+    | {
```

Comment:
What is this `|` about? Looks like a bitwise or but that can't be it, and wrong syntax for pipeline operator. Is it a TypeScript thing?

Reply:
Yeah - called a "type union". The normal syntax is `type X = A | B`, which means every X is either an A or a B (more akin to a logical-or vs. bitwise-or).

TS allows this nice syntax where you can put `| ` at the beginning so you can make a bunch of options line up all nice-like:
```typescript
type X =
  | A
  | B
  | C
  | D
  | E;
```